src/Workspaces/Remote/Core/ServiceDescriptor.cs

```diff
@@ -78,6 +79,26 @@ protected override JsonRpcConnection CreateConnection(JsonRpc jsonRpc)
             return connection;
         }
 
+        public override ServiceRpcDescriptor WithMultiplexingStream(MultiplexingStream? multiplexingStream)
+        {
+            var baseResult = base.WithMultiplexingStream(multiplexingStream);
+            if (baseResult is ServiceDescriptor)
+                return baseResult;
+
+            // work around incorrect implementation in 16.8 Preview 2
+            if (MultiplexingStream == multiplexingStream)
+                return this;
```

Comment:
possibly to doc/explain/link this to something clearer?

Reply:
actually, doc the whole thing tbh, i don't undestand any of it :)

Reply:
See https://github.com/dotnet/roslyn/pull/47870#discussion_r496365307

Pull request #48153 will revert this as soon as roslyn-integration-CI is updated to 16.8 Preview 3 or newer.